File: Count and Say/Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        std::string res;
        if(!n)      return res;
        
        res = "1";
        while(n-- > 1){
            std::string newStr;
            for(unsigned int i = 0; i < res.size();){
                char curr = res[i];
                if(i == res.size() - 1){
                    newStr.push_back('1');
                    newStr.push_back(curr);
                    ++i;
                }
                else{
                    int counter = 1;
                    while(i < res.size()){
                        ++i;
                        if(res[i] == curr){
                            counter++;
                        }
                        else{
                            break;
                        }
                    }
                    char num = counter + '0';
                    newStr.push_back(num);
                    newStr.push_back(curr);
                }
            }
            res = newStr;
        }
        return res;
    }
};

File: Count and Say/Count and Say.java
public class Solution {
    public String countAndSay(int n) {
        
        String s = "1";
        
        while(--n > 0){
            StringBuilder ret = new StringBuilder();
            for(int i = 0; i < s.length();){
                char c = s.charAt(i);
                int curr = c - '0';
                int count = 0;
                while(i < s.length() && s.charAt(i) == c){
                    ++count;
                    ++i;
                }
                String n_char = Integer.toString(count);
                ret.append(n_char);
                ret.append(c);
            }
            s = ret.toString();
        }
        return s;
    }
}
